### DISCUSSION TITLE
I have two questions about Linux Servers with the newest update.

### DISCUSSION THREAD

**Discussion by Orbiszeus** [seleniumbase/SeleniumBase#3110] [Q&A] [answered]
Dear Michael,
1st question: Do we need PyAutoGUI (which version if so) for Linux servers to bypass CF Turnstile?
2nd question: Do I need to say xvfb=True to use the X11 Display because in my previous question I did what you have told me and downloaded the X11 things on my Linux server.

**Comment by mdmintz** [comment, marked as answer]
`PyAutoGUI` is needed for CAPTCHA-bypass on Linux these days. (It is automatically installed if missing when calling a SeleniumBase method that needs to use it.)

`xvfb=True` is needed for the special virtual display on Linux. (Depending on what options you set, that may already be the default option, but it doesn't hurt to add `xvfb=True`.)

Occasionally, CF makes updates to their CAPTCHAs, which may break things temporarily until there's a new SeleniumBase version with a fix. Recently, this has happened frequently. It's called, "The Great CAPTCHA Duel", and I talked about it in UC Mode Video 3: "Revenge of the CAPTCHAs" (https://www.youtube.com/watch?v=-EpZlhGWo9k).
